packages/contracts/contracts/src/BasePolicy.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.27;
+
+import {IBasePolicy} from "./interfaces/IBasePolicy.sol";
+import {Policy} from "./Policy.sol";
+import {BaseChecker} from "./BaseChecker.sol";
+
+/// @title BasePolicy
+/// @notice Abstract base contract which can be extended to implement a specific `BasePolicy`.
+abstract contract BasePolicy is Policy, IBasePolicy {
+    /// @dev Reference to the BaseChecker contract for validation.
+    BaseChecker public immutable BASE_CHECKER;
+
+    /// @dev Tracks whether the check has been enforced for a subject.
+    mapping(address => mapping(address => bool)) public enforced;
```

Comment:
what's the reasoning behind storing addresses as a base? imagine a ERC721 policy, they could transfer the token id to another wallet and pass again? wouldn't the type of data to be stored need to be custom for each policy?

Reply:
The base address-to-address mapping serves as a fundamental tracking mechanism for policy enforcement. While this may seem simple, it's intentionally designed this way for two reasons:

- The policy's primary role is binary - tracking whether a subject has enforced evidence for an address. Complex validation logic (like NFT ownership changes) belongs in policy checker extensions - not at core functionality level (the gatekeeper must enforce and keep track, based on the checks).
- Additional checks and restrictions are implemented at the Checker level, not the Policy level. For example, `AdvancedERC721Checker` under the `test/advanced` folder can track used `tokenIds` to prevent reuse - this is where such specific validation belongs.

This design maintains clean separation between basic enforcement tracking and policy-specific validation rules. Let me know if this makes sense!